src/sentry/preprod/vcs/status_checks/templates.py

```diff
@@ -0,0 +1,165 @@
+        case PreprodArtifact.ArtifactState.UPLOADING | PreprodArtifact.ArtifactState.UPLOADED:
+            summary = _SIZE_ANALYZER_PROCESSING_SUMMARY_TEMPLATE % base_data.__dict__
+            subtitle = str(_SIZE_ANALYZER_SUBTITLE_PROCESSING)
+        case PreprodArtifact.ArtifactState.FAILED:
+            failure_data = _FailureStatusCheckData(
+                build_id=preprod_artifact.id,
+                app_id=preprod_artifact.app_id or "--",
+                version=version_string,
+                error_message=preprod_artifact.error_message or "Unknown error",
+            )
+            summary = _SIZE_ANALYZER_FAILURE_SUMMARY_TEMPLATE % failure_data.__dict__
+            subtitle = str(_SIZE_ANALYZER_SUBTITLE_ERROR)
+        case PreprodArtifact.ArtifactState.PROCESSED:
+            from sentry.preprod.models import PreprodArtifactSizeMetrics
+
+            current_metrics = PreprodArtifactSizeMetrics.objects.filter(
+                preprod_artifact=preprod_artifact,
+                metrics_artifact_type=PreprodArtifactSizeMetrics.MetricsArtifactType.MAIN_ARTIFACT,
+                state=PreprodArtifactSizeMetrics.SizeAnalysisState.COMPLETED,
+            ).first()
+
+            if current_metrics:
+                success_data = _SuccessStatusCheckData(
+                    build_id=preprod_artifact.id,
+                    app_id=preprod_artifact.app_id or "--",
+                    version=version_string,
+                    download_size=_format_file_size(current_metrics.min_download_size),
+                    download_change="N/A",
+                    install_size=_format_file_size(current_metrics.min_install_size),
+                    install_change="N/A",
+                )
+                summary = _SIZE_ANALYZER_SUCCESS_SUMMARY_TEMPLATE % success_data.__dict__
+            else:
+                # TODO(telkins): what to do in this case?
+                summary = (
+                    _("Build %(build_id)s for `%(app_id)s` processed successfully.")
+                    % base_data.__dict__
+                )
+            subtitle = _generate_success_subtitle(preprod_artifact)
+        case _:
+            raise ValueError(f"Invalid artifact state: {preprod_artifact.state}")
+
+    # Convert lazy translation strings to regular strings for JSON serialization
+    return str(title), str(subtitle), str(summary)
+
+
+_SIZE_ANALYZER_TITLE_BASE = _("Size Analysis")
+_SIZE_ANALYZER_SUBTITLE_PROCESSING = _("Processing...")
+_SIZE_ANALYZER_SUBTITLE_SUCCESS = _("Complete")
+_SIZE_ANALYZER_SUBTITLE_ERROR = _("Error processing")
+_SIZE_ANALYZER_SUBTITLE_COMPLETE_FIRST_BUILD = _("1 build analyzed")
+# TODO: Re-add these when size comparison is implemented
+# _SIZE_ANALYZER_SUBTITLE_SIZE_INCREASED = _("1 build increased in size")
+# _SIZE_ANALYZER_SUBTITLE_SIZE_DECREASED = _("1 build decreased in size")
+# _SIZE_ANALYZER_SUBTITLE_NO_CHANGE = _("1 build, no size change")
+
+_SIZE_ANALYZER_PROCESSING_SUMMARY_TEMPLATE = _(
```

Comment:
localizing a template? 😮

Reply:
hopefully the translators don't mess the format up when inputting translated versions